rewards/utils/emission_utils.py

```diff
@@ -0,0 +1,19 @@
+from functools import lru_cache
+from helpers.constants import EMISSIONS_CONTRACTS
+from helpers.web3_utils import make_contract
+
+
+@lru_cache
+def get_emission_control(chain: str):
+    return make_contract(
+        EMISSIONS_CONTRACTS[chain]["EmissionControl"], "EmissionControl", chain
+    )
+
+
+def get_token_weight(token: str, chain: str):
+    token_weight = get_emission_control(chain).tokenWeight(token).call()
+    return 1 if token_weight == 0 else token_weight
+
+
+def get_flat_emission_rate(sett: str, chain: str):
+    return get_emission_control(chain).proRataEmissionRate(sett).call() / 1e4
```

Comment:
What happens here if a vault is not defined on the emissions control?

Reply:
This returns 0, which is default behaviour

Reply:
Looking at the code, the function is:
```
  function proRataEmissionRate(address _vault) external view returns (uint256) {
    return MAX_BPS - boostedEmissionRate[_vault];
  }
  ```
  
  Which means if the vault is not defined, it returns 10000, handled pro-rata, but the function does fail if the address is not checksummed.  We should have the below cases covered:
  - Happy Cases
   - 100% pro rata
   - 100% boosted
   - mixed boosted / pro rata
  - Failing Tests
   - Invalid address format

It's also something @axejintao should note that the default functionality for all addresses not stored in the vaults storage map will have a prorata emissions rate of 100%